### PR TITLE
Negated equality

### PR DESCRIPTION
**Pull request by fredrik-bakke**
I factored out negated equality from #701. Also refactors parts of the library to use this new definition.

### CODE REVIEW

File: src/finite-group-theory/delooping-sign-homomorphism.lagda.md [hunk 1]
```diff
@@ -1512,17 +1513,17 @@ module _
...
-          ( action-equiv-family-over-subuniverse
-            ( mere-equiv-Prop (Fin (n +ℕ 2)))
-            ( type-UU-Fin 2 ∘ Q (n +ℕ 2))
-            ( raise l1 (Fin (n +ℕ 2)) ,
-              unit-trunc-Prop (compute-raise-Fin l1 (n +ℕ 2)))
-            ( raise l1 (Fin (n +ℕ 2)) ,
-              unit-trunc-Prop (compute-raise-Fin l1 (n +ℕ 2)))
-            ( transposition Y))
-          ( x))))
+    x ≠
```
Comment (EgbertRijke) [inline src/finite-group-theory/delooping-sign-homomorphism.lagda.md:1516]:
```suggestion
    ( x) ≠
```

for balanced parentheses

File: src/finite-group-theory/transpositions.lagda.md [hunk 2]
```diff
@@ -466,14 +468,14 @@ module _
...
+    other-element-two-elements-transposition
   neq-elements-two-elements-transposition =
     pr1 (pr2 (pr2 two-elements-transposition))
 
   abstract
     cases-eq-two-elements-transposition :
-      (x y : X) (np : ¬ (Id x y)) →
+      (x y : X) (np : x ≠ y) →
       (type-Decidable-Prop (pr1 Y x)) →
       (type-Decidable-Prop (pr1 Y y)) →
```
Comment (EgbertRijke) [inline src/finite-group-theory/transpositions.lagda.md:480]:
```suggestion
      type-Decidable-Prop (pr1 Y x) →
      type-Decidable-Prop (pr1 Y y) →
```

File: src/finite-group-theory/transpositions.lagda.md [hunk 3]
```diff
@@ -555,7 +557,7 @@ module _
           ( λ p → nq (pr1 (pair-eq-Σ p))))
 
     eq-two-elements-transposition :
-      (x y : X) (np : ¬ (Id x y)) →
+      (x y : X) (np : x ≠ y) →
       (type-Decidable-Prop (pr1 Y x)) →
       (type-Decidable-Prop (pr1 Y y)) →
```
Comment (EgbertRijke) [inline src/finite-group-theory/transpositions.lagda.md:562]:
```suggestion
      type-Decidable-Prop (pr1 Y x) →
      type-Decidable-Prop (pr1 Y y) →
```

File: src/finite-group-theory/transpositions.lagda.md [hunk 4]
```diff
@@ -1231,16 +1233,16 @@ module _
...
   where
 
   cases-htpy-conjugate-transposition :
     (w : X) →
-    ((w ＝ x) + ¬ (w ＝ x)) →
-    ((w ＝ y) + ¬ (w ＝ y)) →
-    ((w ＝ z) + ¬ (w ＝ z)) →
+    ((w ＝ x) + (w ≠ x)) →
+    ((w ＝ y) + (w ≠ y)) →
+    ((w ＝ z) + (w ≠ z)) →
```
Comment (EgbertRijke) [inline src/finite-group-theory/transpositions.lagda.md:1245]:
```suggestion
    ( is-decidable (w ＝ x)) →
    ( is-decidable (w ＝ y)) →
    ( is-decidable (w ＝ z)) →
```

File: src/foundation/apartness-relations.lagda.md [hunk 5]
```diff
@@ -26,16 +26,17 @@ open import foundation-core.propositions
...
-An apartness relation on a type `A` is a relation `R` which is
+An **apartness relation** on a type `A` is a
+[relation](foundation.binary-relations.md) `R` which is
 
-- Antireflexive: For any `a : A` we have `¬ (R a a)`
-- Symmetric: For any `a b : A` we have `R a b → R b a`
-- Cotransitive: For any `a b c : A` we have `R a b → R a c ∨ R b c`.
+- **Antireflexive**: For any `a : A` we have `¬ (R a a)`
+- **Symmetric**: For any `a b : A` we have `R a b → R b a`
+- **Cotransitive**: For any `a b c : A` we have `R a b → R a c ∨ R b c`.
```
Comment (EgbertRijke) [inline src/foundation/apartness-relations.lagda.md:34]:
```suggestion
- **Antireflexive:** For any `a : A` we have `¬ (R a a)`
- **Symmetric:** For any `a b : A` we have `R a b → R b a`
- **Cotransitive:** For any `a b c : A` we have `R a b → R a c ∨ R b c`.
```

File: src/foundation/isolated-elements.lagda.md [hunk 6]
```diff
@@ -75,7 +76,7 @@ module _
 complement-isolated-element :
   {l1 : Level} (X : UU l1) → isolated-element X → UU l1
 complement-isolated-element X x =
-  Σ X (λ y → ¬ (element-isolated-element x ＝ y))
+  Σ X (element-isolated-element x ≠_)
```
Comment (EgbertRijke) [inline src/foundation/isolated-elements.lagda.md:79]: I know that you like the ≠_ and *_ notation for families or functions, but I actually like the lambda-notation:)

In this case the Sigma-type looks less archaic to me with the lambda.

File: src/foundation/multisubsets.lagda.md [hunk 7]
```diff
@@ -57,7 +56,7 @@ module _
   support-locally-finite-multisubset :
     (U : Set l1) → locally-finite-multisubset U → UU l1
   support-locally-finite-multisubset U μ =
-    Σ (type-Set U) λ x → ¬ (μ x ＝ zero-ℕ)
+    Σ (type-Set U) (λ x → μ x ≠ zero-ℕ)
```
Comment (EgbertRijke) [inline src/foundation/multisubsets.lagda.md:59]: Here's a way to make it look even more obscure: `_≠ zero-ℕ ∘ μ`

File: src/foundation/negated-equality.lagda.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+  is-prop-nonequal : {x y : A} → is-prop (x ≠ y)
+  is-prop-nonequal = is-prop-neg
+
+  nonequal-Prop : (x y : A) → Prop l
+  pr1 (nonequal-Prop x y) = x ≠ y
+  pr2 (nonequal-Prop x y) = is-prop-nonequal
+```
+
+### Disequality is antireflexive
```
Comment (EgbertRijke) [inline src/foundation/negated-equality.lagda.md:55]:
```suggestion
### Nonequality is antireflexive
```

File: src/foundation/negated-equality.lagda.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+nonequal : {l : Level} {A : UU l} → A → A → UU l
+nonequal x y = ¬ (x ＝ y)
+
+infix 6 _≠_
+_≠_ = nonequal
+```
+
+## Properties
+
+### Disequality is a property
```
Comment (EgbertRijke) [inline src/foundation/negated-equality.lagda.md:40]:
```suggestion
### Nonequality is a property
```

File: src/foundation/negated-equality.lagda.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+  where
+
+  is-antireflexive-nonequal : (a : A) → ¬ (a ≠ a)
+  is-antireflexive-nonequal a d = d refl
+
+  is-consistent-nonequal : (a b : A) → (a ＝ b) → ¬ (a ≠ b)
+  is-consistent-nonequal a b p d = d p
+```
+
+### Disequality is symmetric
```
Comment (EgbertRijke) [inline src/foundation/negated-equality.lagda.md:69]:
```suggestion
### Nonequality is symmetric
```

File: src/foundation/negated-equality.lagda.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+### If there is any family on `A` that is inhabited over one term but not the other, then they are nonequal
+
+```agda
+module _
+  {l1 l2 : Level} {A : UU l1}
+  where
+
+  nonequal-leibniz : (B : A → UU l2) → (a b : A) → B a → ¬ (B b) → a ≠ b
```
Comment (EgbertRijke) [inline src/foundation/negated-equality.lagda.md:128]: An alternative, more symmetric formulation

```suggestion
  nonequal-leibniz : (B : A → UU l2) → (a b : A) → (B a → B b → empty) → a ≠ b
```

File: src/foundation/repetitions-of-values.lagda.md [hunk 12]
```diff
@@ -86,7 +87,7 @@ module _
 is-repeated-value :
   {l1 l2 : Level} {A : UU l1} {B : UU l2} (f : A → B) (a : A) → UU (l1 ⊔ l2)
 is-repeated-value {l1} {l2} {A} {B} f a =
-  Σ (Σ A (λ x → ¬ (Id a x))) (λ x → f a ＝ f (pr1 x))
+  Σ (Σ A (a ≠_)) (λ x → f a ＝ f (pr1 x))
```
Comment (EgbertRijke) [inline src/foundation/repetitions-of-values.lagda.md:90]: I don't know... to me this is such an agda-specific notation that I would expect it looks strange to the ordinary mathematician. If our goal is that our library uses intuitive notation and conventions for a mathematician, then this seems to go agains that in my opinion.

It's just something I want to bring up, not something I want you to change right now. But perhaps I wouldn't want you to go around the library and change all type families defined using infix binary relations to this form -- that would be something I would like to discuss before we do that.

File: src/foundation/tight-apartness-relations.lagda.md [hunk 13]
```diff
@@ -48,7 +48,7 @@ is-tight-Apartness-Relation R = is-tight (rel-Apartness-Relation R)
 Tight-Apartness-Relation :
   {l1 : Level} (l2 : Level) (A : UU l1) → UU (l1 ⊔ lsuc l2)
 Tight-Apartness-Relation l2 A =
-  Σ (Apartness-Relation l2 A) (λ R → is-tight-Apartness-Relation R)
+  Σ (Apartness-Relation l2 A) (is-tight-Apartness-Relation)
```
Comment (EgbertRijke) [inline src/foundation/tight-apartness-relations.lagda.md:51]:
```suggestion
  Σ (Apartness-Relation l2 A) is-tight-Apartness-Relation
```

File: src/foundation/universal-property-booleans.lagda.md [hunk 14]
```diff
@@ -89,12 +90,12 @@ decide-true-false true = inl refl
 decide-true-false false = inr refl
 
 eq-false :
-  (b : bool) → (¬ (Id b true)) → (Id b false)
+  (b : bool) → (b ≠ true) → (Id b false)
```
Comment (EgbertRijke) [inline src/foundation/universal-property-booleans.lagda.md:93]:
```suggestion
  (b : bool) → b ≠ true → b ＝ false
```

File: src/foundation/universal-property-booleans.lagda.md [hunk 15]
```diff
@@ -89,12 +90,12 @@ decide-true-false true = inl refl
...
 eq-false true p = ind-empty (p refl)
 eq-false false p = refl
 
 eq-true :
-  (b : bool) → (¬ (Id b false)) → Id b true
+  (b : bool) → (b ≠ false) → Id b true
```
Comment (EgbertRijke) [inline src/foundation/universal-property-booleans.lagda.md:98]:
```suggestion
  (b : bool) → b ≠ false → b ＝ true
```

File: src/set-theory/countable-sets.lagda.md [hunk 16]
```diff
@@ -177,8 +178,7 @@ module _
 
   decidable-subprojection-ℕ-enumeration :
     enumeration X → decidable-subprojection-ℕ X
-  pr1 (pr1 (decidable-subprojection-ℕ-enumeration (f , H)) n) =
-    ¬ ((f n) ＝ (inr star))
+  pr1 (pr1 (decidable-subprojection-ℕ-enumeration (f , H)) n) = f n ≠ inr star
```
Comment (EgbertRijke) [inline src/set-theory/countable-sets.lagda.md:181]:
```suggestion
  pr1 (pr1 (decidable-subprojection-ℕ-enumeration (f , H)) n) =
    f n ≠ inr star
```

I like to put them on a new line, to balance them with the other components that go on a new line. I think it looks better.

File: src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md [hunk 17]
```diff
@@ -705,7 +706,7 @@ module _
     ( Y : 2-Element-Decidable-Subtype l X)
     ( two-elements : Σ X
       ( λ x → Σ X
-        ( λ y → Σ (¬ (Id x y))
+        ( λ y → Σ (x ≠ y)
```
Comment (EgbertRijke) [inline src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md:709]:
```suggestion
        ( λ y →
          Σ (x ≠ y)
```

File: src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md [hunk 18]
```diff
@@ -747,7 +748,7 @@ module _
...
```
Comment (EgbertRijke) [inline src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md:751]:
```suggestion
        ( λ y →
          Σ (x ≠ y)
```

Comment (EgbertRijke) [reply src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md:751]: This entire entry might just need to be reformatted properly.

File: src/foundation/negated-equality.lagda.md [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+open import foundation-core.function-extensionality
+open import foundation-core.identity-types
+open import foundation-core.propositions
+```
+
+</details>
+
+## Idea
+
+Two elements `x` and `y` are **not equal** whenever `¬ (x ＝ y)` is inhabited.
```
Comment (EgbertRijke) [inline src/foundation/negated-equality.lagda.md:26]: We should probably enter negated equality in `tables/identity-types.md`

Comment (fredrik-bakke) [reply src/foundation/negated-equality.lagda.md:26]: Ah, thanks

File: src/foundation/tight-apartness-relations.lagda.md (hunk 13 above)
Comment (fredrik-bakke) [reply src/foundation/tight-apartness-relations.lagda.md:51]: I disagree with this suggestion, I find the parentheses improve consistency.

File: src/foundation/repetitions-of-values.lagda.md (hunk 12 above)
Comment (fredrik-bakke) [reply src/foundation/repetitions-of-values.lagda.md]: I agree the syntactic sugar isn't always the correct option so I would be opposed to such a refactoring.

File: src/foundation/tight-apartness-relations.lagda.md (hunk 13 above)
Comment (EgbertRijke) [reply src/foundation/tight-apartness-relations.lagda.md:51]: Ok, feel free to ignore it.

File: src/foundation/multisubsets.lagda.md (hunk 7 above)
Comment (fredrik-bakke) [reply src/foundation/multisubsets.lagda.md:59]: Thanks! I found an even better one though

Comment (EgbertRijke) [reply src/foundation/multisubsets.lagda.md]: Lol, you know that my comment was a joke right?:)

File: src/foundation/multisubsets.lagda.md [hunk 20]
```diff
@@ -57,7 +57,7 @@ module _
   support-locally-finite-multisubset :
     (U : Set l1) → locally-finite-multisubset U → UU l1
   support-locally-finite-multisubset U μ =
-    Σ (type-Set U) λ x → ¬ (μ x ＝ zero-ℕ)
+    Σ (type-Set U) (λ {≠ → precomp μ UU (_≠ 0) ≠})
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/foundation/multisubsets.lagda.md:60]: 🙃

[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/foundation/multisubsets.lagda.md]: Was Agda programmed in Mordor? How is this even valid syntax??

File: src/foundation/negated-equality.lagda.md (hunk 11 above)
Comment (fredrik-bakke) [reply src/foundation/negated-equality.lagda.md:128]: I agree this formulation is better. However, my head is not completely with me today and I don't immediately see how to prove it. Do you have a proof in mind?

File: src/foundation/multisubsets.lagda.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/foundation/multisubsets.lagda.md]: Even worse, it's made by Swedes!

File: src/foundation/negated-equality.lagda.md (hunk 11 above)
Comment (fredrik-bakke) [reply src/foundation/negated-equality.lagda.md:128]: Lol, then I'll resolve this conversation :)

**Comment by EgbertRijke** [comment]
Great! I'll merge when the checks are done.

**Pull request merged**